src/Symfony/Component/Translation/Translator.php

```diff
@@ -119,12 +119,10 @@ public function addLoader(string $format, LoaderInterface $loader)
      *
      * @param string $format   The name of the loader (@see addLoader())
      * @param mixed  $resource The resource name
-     * @param string $locale   The locale
-     * @param string $domain   The domain
      *
      * @throws InvalidArgumentException If the locale contains invalid characters
      */
-    public function addResource(string $format, $resource, string $locale, string $domain = null)
+    public function addResource(string $format, $resource, ?string $locale, string $domain = null)
```

Comment:
I don't see how this could work. The `assertValidLocale()` call below will throw an exception when `null` is pass for the `$locale` argument, won't it?

Reply:
If we look at the test null is a valid locale, maybe the tests are wrong ?

Reply:
assertValidLocale does not seems to throw on null

Reply:
hm indeed: https://3v4l.org/h7sGr

Reply:
But I think we should deprecate the possibility to pass `null` here. Looking at how it is used it doesn't seem like it was intended to be supported.

Reply:
`addResource` and similar methods changed here require a $locale argument which does not have a default null value and  does not implement the fallback to the defaut locale like for example https://github.com/symfony/symfony/blob/ed56c538a1194f4214c0b35a5a572d596ea10751/src/Symfony/Component/Translation/Translator.php#L223-L226
for this reason, I agree that the $locale should not be nullable and calling `$this->resources[$locale]` with null is also strange. so I'm in favor of deprecating null in those places and not making it nullable.

Reply:
Also the locale is nullable in https://github.com/symfony/symfony/blob/ed56c538a1194f4214c0b35a5a572d596ea10751/src/Symfony/Component/Translation/MessageCatalogue.php#L33 but the getter is not nullable. It seems usages of getLocale would also not expect null. So the nullable in the MessageCatalogue constructor should also be deprecated

Reply:
I'm going to do it